examples/panel/weights-and-biases-llm/app.py

```diff
@@ -0,0 +1,207 @@
+    """
+    This function takes a query in natural language
+    and returns a url that can be used to search through 
+    repositories on GitHub
+    """
+    try:
+
+        # start a span to trace the assistant
+        start_time_ms = datetime.datetime.now().timestamp() * 1000
+        run, raw_response = start_assistant(query)
+        messages = as_json(raw_response)
+        json_run = as_json(run)
+        end_time_ms = round(
+                    datetime.datetime.now().timestamp() * 1000
+            ) 
+        # Obtain response from the assistant and log it
+        interpretation = messages['data'][0]['content'][0]['text']['value']
+        status = "success"
+        status_message = (None,)
+        response_text = interpretation
+        token_usage = json_run['usage']
+
+        root_span = trace_log(status, status_message, token_usage, \
+                            start_time_ms, end_time_ms, query, \
+                                response_text, json_run, model_name)
+
+        # log the span to wandb
+        root_span.log(name="openai_trace")
+
+
+    except Exception as e:
+        end_time_ms = round(
+            datetime.datetime.now().timestamp() * 1000
+        )  # logged in milliseconds
+        status = "error"
+        status_message = str(e)
+        response_text = ""
+        token_usage = {}
+
+        root_span = trace_log(status, status_message, token_usage, \
+                            start_time_ms, end_time_ms, query, \
+                                response_text, json_run, model_name)
+
+        # log the span to wandb
+        root_span.log(name="openai_trace")
+        
+    return search_github_repositories(interpretation)
+
+def search_github_repositories(response):
+    # Use regular expressions to extract the URL
+    url_pattern = r"https://api\.github\.com/search/repositories\?.+"
+    match = re.search(url_pattern, response)
+    
+    if match:
+        url = match.group()
+        headers = {
+            "Accept": "application/vnd.github+json",
+            "X-GitHub-Api-Version": "2022-11-28"
+        }
+        response = requests.get(url, headers=headers)
+        json_response = response.json()
+
+        if response.status_code == 200:
+            if len(json_response['items'])<0:
+                return "No repositories found for the given search criteria."
+            if len(json_response['items'])>15:
+                repo_list = "Here are 15 repositories related to your search:\n"
+                repo_list += "\n".join([f"- Description: {json_response['items'][i]['description']}: {json_response['items'][i]['html_url']}. " for i in range(15)])
+                return repo_list
+            else:
+                repo_list = "Here are the repositories related to your search:\n"
+                repo_list += "\n".join([f"- Description: {json_response['items'][i]['description']}: {json_response['items'][i]['html_url']}. " for i in range(len(json_response['items']))])
```

Comment:
the response format is:

- Description: [text] [URL]

this is hard to read, instead we can change it to:

- [name of the repo in bold with the link]: [description]

example:

- [scikit-learn](https://github.com/scikit-learn/scikit-learn): machine learning in Python 

note that the name is clickable and should open a new tab

Reply:
New app deployed

https://damp-scene-8091.ploomberapp.io/app

Resolved https://github.com/ploomber/doc/pull/113/commits/7254ec5af200512001f950b4d2c3a2468b95cb2b

Reply:
this is pretty hard to read, the link should only cover the title, not the whole description

Reply:
also, this error is confusing, it should tell me to provide a valid search term:

![image](https://github.com/ploomber/doc/assets/989250/76655f69-8eef-4e13-a863-1d4884136282)

Reply:
since this is taking too long, let's just fix the links (only in the title) and merge this

Reply:
Displaying url instead of description

Changed app logic so that the llm can provide a better error when given an ambiguous response

https://github.com/ploomber/doc/pull/113/commits/6cf8c111cb69accf160977f2e9f0fafe6af24fa2

Redeployed application 

https://damp-scene-8091.ploomberapp.io/app